### PR TITLE
[FIX] web: propagate can_create into the search popup

### PR DESCRIPTION
**Pull request by kebeclibre**
When creating a SO line and clicking on search more for product_id

Before this commit, every user had the button "Create" to create a product.product

After this commit, only user in groups that actually can create a product have the button

OPW 1910371

Description of the issue/feature this PR addresses:

Current behavior before PR:

Desired behavior after PR is merged:

--
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

### CODE REVIEW

File: addons/web/static/src/js/views/view_dialogs.js [hunk 1]
```diff
@@ -362,7 +362,7 @@ var SelectCreateDialog = ViewDialog.extend({
                 classes: "btn-default o_form_button_cancel",
                 close: true,
             }];
-            if (!self.options.no_create) {
```
Comment (aab-odoo) [inline addons/web/static/src/js/views/view_dialogs.js]: why adding a new option? I'd prefer to use no_create ->
`no_create: !self.can_create,` above

File: addons/web/static/tests/views/view_dialogs_tests.js [hunk 2]
```diff
@@ -420,6 +420,65 @@ QUnit.module('Views', {
...
+        form.$('.o_field_widget[name="instrument"] .o_input').click();
+
+        assert.notOk($('.ui-autocomplete a:contains(Create and Edit)').length,
+            'Create and edit not present in dropdown');
+
+        $('.ui-autocomplete a:contains(Search More)').trigger('mouseenter').trigger('click');
+
+        var $modal = $('.modal-dialog.modal-lg');
+
+        assert.ok($modal.length, 'Modal present');
```
Comment (aab-odoo) [inline addons/web/static/tests/views/view_dialogs_tests.js]: use `assert.containsOnce(document.body, '.modal');` here

Comment (kebeclibre) [reply addons/web/static/tests/views/view_dialogs_tests.js]: @aab-odoo 
I can't: I am working in v11.0 :D

**Comment by kebeclibre** [comment]
@aab-odoo 
changes made

**Comment by aab-odoo** [comment]
robodoo r+

**Pull request closed without merging**